src/asset/asset-registry.js

```diff
@@ -531,6 +531,13 @@ Object.assign(pc, function () {
 
         // private method used for engine-only loading of model data
         _loadMaterials: function (dir, mapping, callback) {
+            if (dir) {
+                dir += '/';
```

Comment:
Should you check that the dir doesn't already end in a forward slash? Two forward slashes together are treated like a jump back to root.

Reply:
In this case, `_loadMaterials` is private and `dir` that is passed to it, is a result of `pc.path.getDirectory(url)` - which returns path without trailing slashes.

Reply:
OK, sounds good, although that's not obvious from the code so I've added some code comments to clarify. Merging now... 😄